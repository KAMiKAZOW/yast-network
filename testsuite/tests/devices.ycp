{

include "testsuite.ycp";

map READ = $[
    "network" : $[
	"section" : $[
	    "arc5"	: nil,
	    "atm5"	: nil,
	    "ci5"	: nil,
	    "ctc5"	: nil,
	    "dummy5"	: nil,
	    "escon5"	: nil,
	    "eth5"	: nil,
	    "eth5:5"	: nil,
	    "eth6"	: nil,
	    "eth7"	: nil,
	    "eth8"	: nil,
	    "eth-pcmcia": nil,
	    "eth-usb"	: nil,
	    "fddi5"	: nil,
	    "hippi5"	: nil,
	    "hsi5"	: nil,
	    "ippp5"	: nil,
	    "iucv5"	: nil,
	    "lo"	: nil,
	    "lo:5"	: nil,
	    "myri5"	: nil,
	    "ppp5"	: nil,
	    "tr5"	: nil,
	    "tr.bak5"	: nil,
	],
	"value" : $[
	    "arc5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "atm5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "ci5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "ctc5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "dummy5"	: $["BOOTPROTO":"static", "IPADDR":"1.2.3.4", "STARTMODE" : "manual"],
	    "escon5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "eth5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "eth5:5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "eth6"	: $["BOOTPROTO":"static", "IPADDR":"1.2.3.4", "STARTMODE":"manual"],
	    "eth7"	: $["STARTMODE":"manual"],
	    "eth8"	: $["IPADDR":"1.2.3.4", "STARTMODE":"manual"],
	    "eth-pcmcia": $["BOOTPROTO":"dhcp", "STARTMODE":"hotplug"],
	    "eth-usb"	: $["BOOTPROTO":"dhcp", "STARTMODE":"hotplug"],
	    "fddi5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "hippi5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "hsi5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "ippp5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "iucv5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "lo"	: $["BROADCAST":"127.255.255.255", "IPADDR":"127.0.0.1", "NETMASK":"255.0.0.0", "NETWORK":"127.0.0.0", "STARTMODE":"onboot"],
	    "lo:5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "myri5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "ppp5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	    "tr5"	: $["BOOTPROTO":"dhcp", "STARTMODE":"manual"],
	]
    ],
    "probe" : $[
	"system" : [],
    ],
    "target" : $[
	"tmpdir" : "/tmp",
    ],
];

map Devices = $[];
list DeletedDevices = [];

TESTSUITE_INIT([READ], nil);

include "network/devices.ycp";

DUMP("ReadDevices");
TEST(``(ReadDevices("")), [READ], nil);
DUMP(sformat("all=%1",Devices));
TEST(``(ReadDevices("eth")), [READ], nil);
DUMP(sformat("eth=%1",Devices));
TEST(``(ReadDevices("ppp")), [READ], nil);
DUMP(sformat("ppp=%1",Devices));
TEST(``(ReadDevices("ippp")), [READ], nil);
DUMP(sformat("ippp=%1",Devices));
TEST(``(ReadDevices("trx")), [READ], nil);
DUMP(sformat("trx=%1",Devices));
TEST(``(ReadDevices("atm|tr")), [READ], nil);
DUMP(sformat("atm|tr=%1",Devices));

DUMP("WriteDevices");
Devices=$["arc":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "atm":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ci":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ctc":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "dummy":$["5":$["BOOTPROTO":"static", "IPADDR":"1.2.3.4", "STARTMODE":"manual"]], "escon":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "eth":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual", "_aliases":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]], "6":$["BOOTPROTO":"static", "IPADDR":"1.2.3.4", "STARTMODE":"manual"], "7":$["STARTMODE":"manual"], "8":$["IPADDR":"1.2.3.4", "STARTMODE":"manual"]], "eth-pcmcia":$["":$["BOOTPROTO":"dhcp", "STARTMODE":"hotplug"]], "eth-usb":$["":$["BOOTPROTO":"dhcp", "STARTMODE":"hotplug"]], "fddi":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "hippi":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "hsi":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ippp":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "iucv":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "lo":$["":$["BROADCAST":"127.255.255.255", "IPADDR":"127.0.0.1", "NETMASK":"255.0.0.0", "NETWORK":"127.0.0.0", "STARTMODE":"onboot", "_aliases":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]]], "myri":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ppp":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "tr":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]];
TEST(``(WriteDevices()), [], nil);

DUMP("GetFreeDevices");
Devices=$["eth":$["0":$[]]];
TEST(``(GetFreeDevices("eth", 2)), [], nil);
Devices=$["eth":$["1":$[]]];
TEST(``(GetFreeDevices("eth", 2)), [], nil);
Devices=$["eth":$["2":$[]]];
TEST(``(GetFreeDevices("eth", 2)), [], nil);
Devices=$["eth-pcmcia":$["":$[]]];
TEST(``(GetFreeDevices("eth-pcmcia", 2)), [], nil);
Devices=$["eth-pcmcia":$["0":$[]]];
TEST(``(GetFreeDevices("eth-pcmcia", 2)), [], nil);
Devices=$["eth-pcmcia":$["1":$[]]];
TEST(``(GetFreeDevices("eth-pcmcia", 2)), [], nil);

DUMP("LocateDevices");
Devices=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "dhcp" ]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
Devices=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "" ]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
Devices=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "static" ]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
Devices=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "static" ], "1" : $[ "BOOTPROTO" : "dhcp"]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);
Devices=$[ "eth" : $[ "0" : $[ "BOOTPROTO" : "static" ]], "tr" : $["1" : $[ "BOOTPROTO" : "dhcp"]]];
TEST(``(LocateDevices("BOOTPROTO", "dhcp")), [], nil);

DUMP("UpdateModemSymlink");
Devices=$["arc":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]];
TEST(``(UpdateModemSymlink()), [], nil);
Devices=$["ppp":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]];
TEST(``(UpdateModemSymlink()), [], nil);
Devices=$["ppp":$["5":$["MODEM_DEVICE":"", "STARTMODE":"manual"]]];
TEST(``(UpdateModemSymlink()), [], nil);
Devices=$["ppp":$["5":$["MODEM_DEVICE":"/dev/modem", "STARTMODE":"manual"]]];
TEST(``(UpdateModemSymlink()), [], nil);
Devices=$["ppp":$["5":$["MODEM_DEVICE":"/dev/ttyS1", "STARTMODE":"manual"]]];
TEST(``(UpdateModemSymlink()), [], nil);
READ=$["target":$["lstat":$["islink":true], "symlink":"/dev/ttyS1"]];
TEST(``(UpdateModemSymlink()), [READ], nil);
READ=$["target":$["lstat":$["islink":true], "symlink":"/dev/ttyS2"]];
TEST(``(UpdateModemSymlink()), [READ], nil);

}
