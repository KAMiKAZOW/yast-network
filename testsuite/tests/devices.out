Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Read	.target.size "/etc/install.inf" nil
Read	.etc.install_inf.Sourcemounted nil
Read	.etc.install_inf.AutoYaST nil
Read	.etc.install_inf.Repair nil
Read	.etc.install_inf.Demo nil
Read	.etc.install_inf.Textmode nil
Read	.etc.install_inf.Manual nil
Read	.target.string "/proc/cmdline" nil
Read	.etc.install_inf.Console nil
Read	.etc.install_inf.Braille nil
Read	.etc.install_inf.VNC nil
Read	.etc.install_inf.UseSSH nil
Read	.etc.install_inf.InstMode nil
Read	.target.tmpdir "/tmp"
Dump	ReadDevices
Dir	.network.section: ["arc5", "atm5", "ci5", "ctc5", "dummy5", "escon5", "eth-pcmcia", "eth-usb", "eth5", "eth6", "eth7", "eth8", "fddi5", "hippi5", "hsi5", "ippp5", "iucv5", "lo", "myri5", "ppp5", "tr.bak5", "tr5"]
Dir	.network.value."arc5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."arc5".BOOTPROTO "dhcp"
Read	.network.value."arc5".STARTMODE "manual"
Dir	.network.value."atm5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."atm5".BOOTPROTO "dhcp"
Read	.network.value."atm5".STARTMODE "manual"
Dir	.network.value."ci5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."ci5".BOOTPROTO "dhcp"
Read	.network.value."ci5".STARTMODE "manual"
Dir	.network.value."ctc5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."ctc5".BOOTPROTO "dhcp"
Read	.network.value."ctc5".STARTMODE "manual"
Dir	.network.value."dummy5": ["BOOTPROTO", "IPADDR", "STARTMODE"]
Read	.network.value."dummy5".BOOTPROTO "static"
Read	.network.value."dummy5".IPADDR "1.2.3.4"
Read	.network.value."dummy5".STARTMODE "manual"
Dir	.network.value."escon5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."escon5".BOOTPROTO "dhcp"
Read	.network.value."escon5".STARTMODE "manual"
Dir	.network.value."eth-pcmcia": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."eth-pcmcia".BOOTPROTO "dhcp"
Read	.network.value."eth-pcmcia".STARTMODE "hotplug"
Dir	.network.value."eth-usb": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."eth-usb".BOOTPROTO "dhcp"
Read	.network.value."eth-usb".STARTMODE "hotplug"
Dir	.network.value."eth5": ["BOOTPROTO", "IPADDR_x", "NETMASK_x", "STARTMODE"]
Read	.network.value."eth5".BOOTPROTO "dhcp"
Read	.network.value."eth5".IPADDR_x "1.1.1.1"
Read	.network.value."eth5".NETMASK_x "0.0.0.0"
Read	.network.value."eth5".STARTMODE "manual"
Dir	.network.value."eth6": ["BOOTPROTO", "IPADDR", "STARTMODE"]
Read	.network.value."eth6".BOOTPROTO "static"
Read	.network.value."eth6".IPADDR "1.2.3.4"
Read	.network.value."eth6".STARTMODE "manual"
Dir	.network.value."eth7": ["STARTMODE"]
Read	.network.value."eth7".STARTMODE "manual"
Dir	.network.value."eth8": ["IPADDR", "STARTMODE"]
Read	.network.value."eth8".IPADDR "1.2.3.4"
Read	.network.value."eth8".STARTMODE "manual"
Dir	.network.value."fddi5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."fddi5".BOOTPROTO "dhcp"
Read	.network.value."fddi5".STARTMODE "manual"
Dir	.network.value."hippi5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."hippi5".BOOTPROTO "dhcp"
Read	.network.value."hippi5".STARTMODE "manual"
Dir	.network.value."hsi5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."hsi5".BOOTPROTO "dhcp"
Read	.network.value."hsi5".STARTMODE "manual"
Dir	.network.value."ippp5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."ippp5".BOOTPROTO "dhcp"
Read	.network.value."ippp5".STARTMODE "manual"
Dir	.network.value."iucv5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."iucv5".BOOTPROTO "dhcp"
Read	.network.value."iucv5".STARTMODE "manual"
Dir	.network.value."lo": ["BROADCAST", "IPADDR", "IPADDR_1", "NETMASK", "NETWORK", "STARTMODE"]
Read	.network.value."lo".BROADCAST "127.255.255.255"
Read	.network.value."lo".IPADDR "127.0.0.1"
Read	.network.value."lo".IPADDR_1 "7.7.7.7"
Read	.network.value."lo".NETMASK "255.0.0.0"
Read	.network.value."lo".NETWORK "127.0.0.0"
Read	.network.value."lo".STARTMODE "onboot"
Dir	.network.value."myri5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."myri5".BOOTPROTO "dhcp"
Read	.network.value."myri5".STARTMODE "manual"
Dir	.network.value."ppp5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."ppp5".BOOTPROTO "dhcp"
Read	.network.value."ppp5".STARTMODE "manual"
Dir	.network.value."tr5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."tr5".BOOTPROTO "dhcp"
Read	.network.value."tr5".STARTMODE "manual"
Return	true
Dump	all=$["arc":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "atm":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ci":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ctc":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "dummy":$["5":$["BOOTPROTO":"static", "IPADDR":"1.2.3.4", "STARTMODE":"manual"]], "escon":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "eth":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual", "_aliases":$["x":$["IPADDR":"1.1.1.1", "NETMASK":"0.0.0.0"]]], "6":$["BOOTPROTO":"static", "IPADDR":"1.2.3.4", "STARTMODE":"manual"], "7":$["STARTMODE":"manual"], "8":$["IPADDR":"1.2.3.4", "STARTMODE":"manual"]], "eth-pcmcia":$["":$["BOOTPROTO":"dhcp", "STARTMODE":"hotplug"]], "eth-usb":$["":$["BOOTPROTO":"dhcp", "STARTMODE":"hotplug"]], "fddi":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "hippi":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "hsi":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ippp":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "iucv":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "lo":$["":$["BROADCAST":"127.255.255.255", "IPADDR":"127.0.0.1", "NETMASK":"255.0.0.0", "NETWORK":"127.0.0.0", "STARTMODE":"onboot", "_aliases":$["1":$["IPADDR":"7.7.7.7"]]]], "myri":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "ppp":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "tr":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]]
Dir	.network.section: ["arc5", "atm5", "ci5", "ctc5", "dummy5", "escon5", "eth-pcmcia", "eth-usb", "eth5", "eth6", "eth7", "eth8", "fddi5", "hippi5", "hsi5", "ippp5", "iucv5", "lo", "myri5", "ppp5", "tr.bak5", "tr5"]
Dir	.network.value."eth5": ["BOOTPROTO", "IPADDR_x", "NETMASK_x", "STARTMODE"]
Read	.network.value."eth5".BOOTPROTO "dhcp"
Read	.network.value."eth5".IPADDR_x "1.1.1.1"
Read	.network.value."eth5".NETMASK_x "0.0.0.0"
Read	.network.value."eth5".STARTMODE "manual"
Dir	.network.value."eth6": ["BOOTPROTO", "IPADDR", "STARTMODE"]
Read	.network.value."eth6".BOOTPROTO "static"
Read	.network.value."eth6".IPADDR "1.2.3.4"
Read	.network.value."eth6".STARTMODE "manual"
Dir	.network.value."eth7": ["STARTMODE"]
Read	.network.value."eth7".STARTMODE "manual"
Dir	.network.value."eth8": ["IPADDR", "STARTMODE"]
Read	.network.value."eth8".IPADDR "1.2.3.4"
Read	.network.value."eth8".STARTMODE "manual"
Return	true
Dump	eth=$["eth":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual", "_aliases":$["x":$["IPADDR":"1.1.1.1", "NETMASK":"0.0.0.0"]]], "6":$["BOOTPROTO":"static", "IPADDR":"1.2.3.4", "STARTMODE":"manual"], "7":$["STARTMODE":"manual"], "8":$["IPADDR":"1.2.3.4", "STARTMODE":"manual"]]]
Dir	.network.section: ["arc5", "atm5", "ci5", "ctc5", "dummy5", "escon5", "eth-pcmcia", "eth-usb", "eth5", "eth6", "eth7", "eth8", "fddi5", "hippi5", "hsi5", "ippp5", "iucv5", "lo", "myri5", "ppp5", "tr.bak5", "tr5"]
Dir	.network.value."ppp5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."ppp5".BOOTPROTO "dhcp"
Read	.network.value."ppp5".STARTMODE "manual"
Return	true
Dump	ppp=$["ppp":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]]
Dir	.network.section: ["arc5", "atm5", "ci5", "ctc5", "dummy5", "escon5", "eth-pcmcia", "eth-usb", "eth5", "eth6", "eth7", "eth8", "fddi5", "hippi5", "hsi5", "ippp5", "iucv5", "lo", "myri5", "ppp5", "tr.bak5", "tr5"]
Dir	.network.value."ippp5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."ippp5".BOOTPROTO "dhcp"
Read	.network.value."ippp5".STARTMODE "manual"
Return	true
Dump	ippp=$["ippp":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]]
Dir	.network.section: ["arc5", "atm5", "ci5", "ctc5", "dummy5", "escon5", "eth-pcmcia", "eth-usb", "eth5", "eth6", "eth7", "eth8", "fddi5", "hippi5", "hsi5", "ippp5", "iucv5", "lo", "myri5", "ppp5", "tr.bak5", "tr5"]
Return	true
Dump	trx=$[]
Dir	.network.section: ["arc5", "atm5", "ci5", "ctc5", "dummy5", "escon5", "eth-pcmcia", "eth-usb", "eth5", "eth6", "eth7", "eth8", "fddi5", "hippi5", "hsi5", "ippp5", "iucv5", "lo", "myri5", "ppp5", "tr.bak5", "tr5"]
Dir	.network.value."atm5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."atm5".BOOTPROTO "dhcp"
Read	.network.value."atm5".STARTMODE "manual"
Dir	.network.value."tr5": ["BOOTPROTO", "STARTMODE"]
Read	.network.value."tr5".BOOTPROTO "dhcp"
Read	.network.value."tr5".STARTMODE "manual"
Return	true
Dump	atm|tr=$["atm":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]], "tr":$["5":$["BOOTPROTO":"dhcp", "STARTMODE":"manual"]]]
Dump	WriteDevices
Write	.network.value."arc5".BOOTPROTO "dhcp" true
Write	.network.value."arc5".STARTMODE "manual" true
Write	.network.value."atm5".BOOTPROTO "dhcp" true
Write	.network.value."atm5".STARTMODE "manual" true
Write	.network.value."ci5".BOOTPROTO "dhcp" true
Write	.network.value."ci5".STARTMODE "manual" true
Write	.network.value."ctc5".BOOTPROTO "dhcp" true
Write	.network.value."ctc5".STARTMODE "manual" true
Write	.network.value."dummy5".BOOTPROTO "static" true
Write	.network.value."dummy5".IPADDR "1.2.3.4" true
Write	.network.value."dummy5".STARTMODE "manual" true
Write	.network.value."escon5".BOOTPROTO "dhcp" true
Write	.network.value."escon5".STARTMODE "manual" true
Write	.network.value."eth5".BOOTPROTO "dhcp" true
Write	.network.value."eth5".STARTMODE "manual" true
Write	.network.value."eth5".IPADDR_x "1.1.1.1" true
Write	.network.value."eth5".NETMASK_x "0.0.0.0" true
Write	.network.value."eth6".BOOTPROTO "static" true
Write	.network.value."eth6".IPADDR "1.2.3.4" true
Write	.network.value."eth6".STARTMODE "manual" true
Write	.network.value."eth7".STARTMODE "manual" true
Write	.network.value."eth8".IPADDR "1.2.3.4" true
Write	.network.value."eth8".STARTMODE "manual" true
Write	.network.value."eth-pcmcia".BOOTPROTO "dhcp" true
Write	.network.value."eth-pcmcia".STARTMODE "hotplug" true
Write	.network.value."eth-usb".BOOTPROTO "dhcp" true
Write	.network.value."eth-usb".STARTMODE "hotplug" true
Write	.network.value."fddi5".BOOTPROTO "dhcp" true
Write	.network.value."fddi5".STARTMODE "manual" true
Write	.network.value."hippi5".BOOTPROTO "dhcp" true
Write	.network.value."hippi5".STARTMODE "manual" true
Write	.network.value."hsi5".BOOTPROTO "dhcp" true
Write	.network.value."hsi5".STARTMODE "manual" true
Write	.network.value."ippp5".BOOTPROTO "dhcp" true
Write	.network.value."ippp5".STARTMODE "manual" true
Write	.network.value."iucv5".BOOTPROTO "dhcp" true
Write	.network.value."iucv5".STARTMODE "manual" true
Write	.network.value."lo".BROADCAST "127.255.255.255" true
Write	.network.value."lo".IPADDR "127.0.0.1" true
Write	.network.value."lo".NETMASK "255.0.0.0" true
Write	.network.value."lo".NETWORK "127.0.0.0" true
Write	.network.value."lo".STARTMODE "onboot" true
Write	.network.value."lo".IPADDR_1 "7.7.7.7" true
Write	.network.value."myri5".BOOTPROTO "dhcp" true
Write	.network.value."myri5".STARTMODE "manual" true
Write	.network.value."ppp5".BOOTPROTO "dhcp" true
Write	.network.value."ppp5".STARTMODE "manual" true
Write	.network.value."tr5".BOOTPROTO "dhcp" true
Write	.network.value."tr5".STARTMODE "manual" true
Write	.network nil true
Return	true
Dump	GetFreeDevices
Return	["1", "2"]
Return	["0", "2"]
Return	["0", "1"]
Return	["0", "1"]
Return	["", "1"]
Return	["", "0"]
Dump	LocateDevices
Return	["eth0"]
Return	[]
Return	[]
Return	["eth1"]
Return	["tr1"]
Dump	UpdateModemSymlink
Return	false
Return	false
Return	false
Return	false
Read	.target.lstat "/dev/modem" nil
Execute	.target.symlink "/dev/ttyS1" "/dev/modem" 0
Return	true
Read	.target.lstat "/dev/modem" $["islink":true]
Read	.target.symlink "/dev/modem" "/dev/ttyS1"
Return	false
Read	.target.lstat "/dev/modem" $["islink":true]
Read	.target.symlink "/dev/modem" "/dev/ttyS2"
Execute	.target.symlink "/dev/ttyS1" "/dev/modem" 0
Return	true
